io/post_fv3.F90

```diff
@@ -3085,94 +3743,677 @@ subroutine set_postvars_fv3(wrt_int_state,grid_id,mype,mpicomp)
+              sulfcb(i,j)=sulfcb(i,j)+ suso(i,j,l,1)* &
+              dpres(i,j,l)/grav
+              endif
+              if(pp25(i,j,l,1)<spval)then
+              pp25cb(i,j)=pp25cb(i,j)+ pp25(i,j,l,1)* &
+              dpres(i,j,l)/grav
+              endif
+              if(pp10(i,j,l,1)<spval)then
+              pp10cb(i,j)=pp10cb(i,j)+ pp10(i,j,l,1)* &
+              dpres(i,j,l)/grav
+              endif
+            enddo
+          enddo
+        end do ! do loop for l
+
+        l=lm
+        do j=jsta,jend
+          do i=ista,iend
+
+              TV = T(I,J,L) * (H1+D608*MAX(Q(I,J,L),QMIN))
+              RHOMID(I,J,L) = PMID(I,J,L) / (RD*TV)
+
+            dustcb(i,j) = MAX(dustcb(i,j), 0.0)
+            dustallcb(i,j) = MAX(dustallcb(i,j), 0.0)
+            sscb(i,j) = MAX(sscb(i,j), 0.0)
+            ssallcb(i,j) = MAX(ssallcb(i,j), 0.0)
+            bccb(i,j) = MAX(bccb(i,j), 0.0)
+            occb(i,j) = MAX(occb(i,j), 0.0)
+            sulfcb(i,j) = MAX(sulfcb(i,j), 0.0)
+            if (nasa_on) then
+            no3cb(i,j) = MAX(no3cb(i,j), 0.0)
+            nh4cb(i,j) = MAX(nh4cb(i,j), 0.0)
+            endif
+            pp25cb(i,j) = MAX(pp25cb(i,j), 0.0)
+            pp10cb(i,j) = MAX(pp10cb(i,j), 0.0)
+
+!      Surface PM25 dust and seasalt      
+           dustpm(i,j)=(dust(i,j,l,1)+0.38*dust(i,j,l,2))*RHOMID(i,j,l) !ug/m3
+           dustpm10(i,j)=(dust(i,j,l,1)+dust(i,j,l,2)+dust(i,j,l,3)+ &
+            0.74*dust(i,j,l,4))*RHOMID(i,j,l) !ug/m3
+           sspm(i,j)=(salt(i,j,l,1)+salt(i,j,l,2)+ &
+            0.83*salt(i,j,l,3))*RHOMID(i,j,l)  !ug/m3 
+
+           if (gocart_on .or. gccpp_on) then
+!      Surface PM10 concentration
+           dusmass(i,j)=(dust(i,j,l,1)+dust(i,j,l,2)+dust(i,j,l,3)+ &
+            0.74*dust(i,j,l,4)+salt(i,j,l,1)+salt(i,j,l,2)+salt(i,j,l,3)+ &
+            salt(i,j,l,4) + soot(i,j,l,1)+soot(i,j,l,2)+waso(i,j,l,1)+ &
+            waso(i,j,l,2) +suso(i,j,l,1)+pp25(i,j,l,1)+pp10(i,j,l,1)) &
+            *RHOMID(i,j,l)  !ug/m3
+!      Surface PM25 concentration       
+           dusmass25(i,j)=(dust(i,j,l,1)+0.38*dust(i,j,l,2)+ &
+           salt(i,j,l,1)+salt(i,j,l,2)+0.83*salt(i,j,l,3) + &
+           soot(i,j,l,1)+soot(i,j,l,2)+waso(i,j,l,1)+ &
+           waso(i,j,l,2) +suso(i,j,l,1)+pp25(i,j,l,1))*RHOMID(i,j,l)  !ug/m3
+
+!      PM10 column
+           ducmass(i,j)=dustallcb(i,j)+ssallcb(i,j)+bccb(i,j)+ &
+           occb(i,j)+sulfcb(i,j)+pp25cb(i,j)+pp10cb(i,j)
+!      PM25 column
+           ducmass25(i,j)=dustcb(i,j)+sscb(i,j)+bccb(i,j)+occb(i,j) &
+            +sulfcb(i,j)+pp25cb(i,j)
+           endif !gocart_on or gccpp_on
+
+           if (nasa_on) then
+!      Surface PM10 concentration
+           dusmass(i,j)=pp10(i,j,l,1)*RHOMID(i,j,l)  !ug/m3
+!      Surface PM25 concentration       
+           dusmass25(i,j)=pp25(i,j,l,1)*RHOMID(i,j,l)  !ug/m3
+
+!      PM10 column
+           ducmass(i,j)=pp10cb(i,j)
+!      PM25 column
+           ducmass25(i,j)=pp25cb(i,j)
```

Comment:
dusmass, dusmass25, ducmass and ducmass25 are computed both in if (gocart_on .or. gccpp_on) and if (nasa_on). 
If both gocart_on and nasa_on are true then second compuation will overwrite the first one. Can they both be true at the same time? If not then this should be `if ... else if .. end if` not two separate if blocks.

Reply:
@DusanJovic-NOAA It could be both gocart_on and nasa_on are true. In that case it is intention second calculation overwrite the first one. @zhanglikate Can you confirm that?

Reply:
gocart_on (or gccpp_on),  nasa_on, will never be  true at the same time since we can only run  one chemical model at one time. Thanks. @WenMeng-NOAA @DusanJovic-NOAA

Reply:
@zhanglikate Thanks for clarifying. I will change based on @DusanJovic-NOAA 's suggestion` if ... else if .. end if`